pkg/cloud/services/autoscaling/autoscalinggroup.go

```diff
@@ -465,3 +460,40 @@ func mapToTags(input map[string]string, resourceID *string) []*autoscaling.Tag {
 	}
 	return tags
 }
+
+// SubnetIDs return subnet IDs of a AWSMachinePool based on given subnetIDs and filters.
+func (s *Service) SubnetIDs(scope *scope.MachinePoolScope) ([]string, error) {
+	subnetIDs := make([]string, 0)
+	var inputFilters = make([]*ec2.Filter, 0)
+
+	for _, subnet := range scope.AWSMachinePool.Spec.Subnets {
+		switch {
+		case subnet.ID != nil && subnet.Filters != nil:
```

Comment:
Can we add this validation to webhook validation so that flow doesn't have to reach till the actual ASG code?